test/unit/connectors/ApplicationConnectorSpec.scala

```diff
@@ -109,10 +109,31 @@ class ApplicationConnectorSpec extends UnitSpec with Matchers with ScalaFutures
     "propagate FetchApplicationsFailed exception" in new Setup {
       stubFor(get(urlEqualTo(s"/gatekeeper/applications")).willReturn(aResponse().withStatus(500)))
 
-      intercept[FetchApplicationsFailed](await(connector.fetchApplications()))
+      intercept[FetchApplicationsFailed](await(connector.fetchApplicationsWithUpliftRequest()))
 
       verify(1, getRequestedFor(urlPathEqualTo(s"/gatekeeper/applications"))
         .withHeader("Authorization", equalTo(authToken)))
     }
   }
+
+  "fetchAllApplicationsBySubscription" should {
+    "retrieve all applications subscribed to a specific API" in new Setup {
+      stubFor(get(urlEqualTo(s"/application?subscribesTo=some-context")).willReturn(aResponse().withStatus(200)
+        .withBody("[]")))
+
+      val result = await(connector.fetchAllApplicationsBySubscription(Some("some-context")))
+
+      verify(1, getRequestedFor(urlPathEqualTo("/application?subscribesTo=some-context"))
+        .withHeader("Authorization", equalTo(authToken)))
+    }
```

Comment:
There appears to be a missing test cases for the `None` context - although as noted above, I think that would be better as another method on the connector.